File: Programmers/Level2/H-Index/H-Index.py
def solution(citations):
    for h in range(max(citations), 0, -1):
        up_cnt = 0
        for num in citations:
            if num >= h:
                up_cnt += 1
            if up_cnt == h:
                return h
    return 0

citations = [3, 0, 6, 1, 5];
print(solution(citations))

File: Programmers/Level2/H-Index/H-Index.js
function solution(citations) {
  const maxVal = Math.max.apply(null, citations);

  for (let i = maxVal; i >= 0; i--) {
    let cnt = 0;
    for (let j = 0; j < citations.length; j++) {
      if (citations[j] >= i) {
        cnt += 1;
      }
      if (cnt === i) return i;
    }
  }
  return 0;
}

const citations = [3, 0, 6, 1, 5];
console.log(solution(citations));
